File: record-generator/processor/src/main/java/me/bristermitten/mittenlib/codegen/union/ResolvedUnionSpec.java
package me.bristermitten.mittenlib.codegen.union;

import com.squareup.javapoet.ClassName;
import java.util.List;
import me.bristermitten.mittenlib.codegen.GenericTypeSpec;
import me.bristermitten.mittenlib.codegen.MatchStrategies;

public record ResolvedUnionSpec(
        ClassName source, ClassName name, MatchStrategies strategy, List<ResolvedUnionConstructor> constructors)
        implements GenericTypeSpec {}

File: record-generator/processor/src/main/scala/me/bristermitten/mittenlib/codegen/union/ResolvedUnionSpec.scala
package me.bristermitten.mittenlib.codegen.union

import com.palantir.javapoet.ClassName
import java.util.List
import me.bristermitten.mittenlib.codegen.GenericTypeSpec
import me.bristermitten.mittenlib.codegen.MatchStrategies

case class ResolvedUnionSpec(
  source: ClassName,
  name: ClassName,
  strategy: MatchStrategies,
  constructors: List[ResolvedUnionConstructor]
) extends GenericTypeSpec
